lib/MiLight/V2PacketFormatter.cpp

```diff
@@ -80,3 +80,23 @@ void V2PacketFormatter::format(uint8_t const* packet, char* buffer) {
 uint8_t V2PacketFormatter::groupCommandArg(MiLightStatus status, uint8_t groupId) {
   return GROUP_COMMAND_ARG(status, groupId, numGroups);
 }
+
+// helper method to return a bulb to the prior state
+void V2PacketFormatter::switchMode(GroupState currentState, BulbMode desiredMode) {
```

Comment:
Perhaps we should skip doing an update if `currentState.getBulbMode() == desiredMode`?

Reply:
This doesn't work because the current state isn't changed right away, causing the comparison to be invalid.  I believe it doesn't change until the packet round-trips, which is after the inline code executes.